README.md

```diff
@@ -48,11 +48,45 @@ secret key used to authenticate the session. Inside the handler, we call
 session values in session.Values, which is a `map[interface{}]interface{}`.
 And finally we call `session.Save()` to save the session in the response.
 
-Important Note: If you aren't using gorilla/mux, you need to wrap your handlers
-with
+## Avoiding Memory Leaks
+
+If you are not using gorilla/mux (or are using context), you will
+need to perform additional cleanup or else you will leak memory!
+
+Here is an example that illustrates proper cleanup:
+
+```go
+	import (
+		"context"
+		"net/http"
+
+		gorillacontext "github.com/gorilla/context"
+		"github.com/gorilla/sessions"
+	)
+
+	var store = sessions.NewCookieStore([]byte("something-very-secret"))
+
+	func MyHandler(w http.ResponseWriter, r *http.Request) {
+		// Store some context on the request. This creates
+		// a shallow copy of the request object.
+		r = r.WithContext(context.TODO())
+
+		// Get a session. We're ignoring the error resulted from decoding an
+		// existing session: Get() always returns a session, even if empty.
+		session, _ := store.Get(r, "session-name")
+
+		// Clear the registry at the end of the request lifecycle
+		defer gorillacontext.Clear(r)
```

Comment:
This doesn't seem correct: this should be done at the router level as middleware - e.g. `http.ListenAndServe(":8080", context.ClearHandler(http.DefaultServeMux))` as per http://www.gorillatoolkit.org/pkg/sessions#overview

Reply:
https://github.com/gorilla/sessions/issues/122

I ran into the same issue where `ClearHandler` is not preventing memory leak. We don't want to mix gorilla/context with context.context but we are forced to because we want to use this library (sessions).

Reply:
I wrote [_seh/handler_](https://godoc.org/github.com/seh/handler) to work around this problem with the registry.